### PR TITLE
Don't initialize the software selection if the payload is not set up (#1916114)

### PR DESCRIPTION
**Pull request by poncovka**
We reset the DNF base if the payload cannot be set up, so it is not ready for
the initialization and validation of the software selection.

Resolves: rhzb#1916114

### CODE REVIEW

**Review by rvykydal — APPROVED** [review]
Looks good to me.

**Review by VladimirSlavik — APPROVED** [review]
Looks good to me. Thank you!

File: pyanaconda/ui/tui/spokes/software_selection.py [hunk 1]
```diff
@@ -198,11 +198,6 @@ def refresh(self, args=None):
         threadMgr.wait(THREAD_PAYLOAD)
         self._container = None
 
-        if not self.payload.base_repo:
-            message = TextWidget(_("Installation source needs to be set up first."))
```
Comment (jkonecny12) [inline pyanaconda/ui/tui/spokes/software_selection.py:202]: Wouldn't users miss this message?

**Review by jkonecny12 — APPROVED** [review]
Looks good to me. I just have one question below.

File: pyanaconda/ui/tui/spokes/software_selection.py (hunk 1 above)
Comment (poncovka) [reply pyanaconda/ui/tui/spokes/software_selection.py]: Good point. I didn't realize that, in TUI, you can enter spokes that are not ready. Fixed.

**Pull request merged**